packages/compass/src/main/window-manager.ts

```diff
@@ -265,6 +272,8 @@ class CompassWindowManager {
         return getWindowAutoConnectPreferences(bw, compassApp.preferences);
       },
       'test:show-connect-window': () => showConnectWindow(compassApp),
+      'app:connect-in-new-window': (event, connectionId: string) =>
+        showConnectWindow(compassApp, { connectionId }),
```

Comment:
These listeners will try to return value back to renderer, if value is not serialized, you'll get an error. Seems like an existing issue that you just copied, we probably want to fix it for both cases
```suggestion
      'test:show-connect-window': () => void showConnectWindow(compassApp),
      'app:connect-in-new-window': (event, connectionId: string) =>
        void showConnectWindow(compassApp, { connectionId }),
```